src/figuresClasses.ts

```diff
@@ -1,19 +1,92 @@
-export interface Figure {
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle'
+}
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape;
 
-}
+  getArea(): number {
+    const half = (this.a
+      + this.b
+      + this.c)
+      / 2;
+
+    return Math.floor((Math.sqrt(
+      half
+      * (half - this.a)
+      * (half - this.b)
+      * (half - this.c),
+    )
+      * 100)) / 100;
+  }
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('all values have to be greater than 0');
+    }
 
-export class Circle {
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('not triangle');
+    }
 
+    this.shape = Shape.triangle;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape;
```

Comment:
You didn't need to remove it completety but now you have it repeated twice, here and in the constructor. If you want to keep it here you can remove it from the constructor. Or the other way around, just don't repeat it